GVFS/GVFS.UnitTests/Maintenance/LooseObjectStepTests.cs

```diff
@@ -61,21 +63,84 @@ public void LooseObjectsPassTimeRestriction()
             this.tracer.StartActivityTracer.RelatedErrorEvents.Count.ShouldEqual(0);
             this.tracer.StartActivityTracer.RelatedWarningEvents.Count.ShouldEqual(0);
             List<string> commands = this.gitProcess.CommandsRun;
-            commands.Count.ShouldEqual(1);
+            commands.Count.ShouldEqual(2);
             commands[0].ShouldEqual(PrunePackedCommand);
+            commands[1].ShouldEqual(this.packCommand);
+        }
+
+        [TestCase]
+        public void LooseObjectsLimitPackCount()
+        {
+            this.TestSetup(DateTime.UtcNow.AddDays(-7));
+
+            // Verify with default limit
+            LooseObjectsStep step = new LooseObjectsStep(this.context, requireCacheLock: false, forceRun: false);
+            step.PackLooseObjects(new StreamWriter(new MemoryStream()));
+            step.LooseObjectsPutIntoPackFile.ShouldEqual(3);
+
+            // Verify with limit of 2
+            step.MaxLooseObjectsInPack = 2;
+            step.PackLooseObjects(new StreamWriter(new MemoryStream()));
+            step.LooseObjectsPutIntoPackFile.ShouldEqual(2);
+        }
+
+        [TestCase]
+        public void SkipInvalidLooseObjects()
+        {
+            this.TestSetup(DateTime.UtcNow.AddDays(-7));
+
+            // Verify with valid Objects 
+            LooseObjectsStep step = new LooseObjectsStep(this.context, requireCacheLock: false, forceRun: false);
+            step.PackLooseObjects(new StreamWriter(new MemoryStream()));
+            step.LooseObjectsPutIntoPackFile.ShouldEqual(3);
+            this.tracer.RelatedErrorEvents.Count.ShouldEqual(0);
+            this.tracer.RelatedWarningEvents.Count.ShouldEqual(0);
+
+            // Write an ObjectId file with an invalid name
+            this.context.FileSystem.WriteAllText(Path.Combine(this.context.Enlistment.GitObjectsRoot, "AA", "NOT_A_SHA"), string.Empty);
+
+            // Verify it wasn't added and a warning exists
+            step.PackLooseObjects(new StreamWriter(new MemoryStream()));
+            step.LooseObjectsPutIntoPackFile.ShouldEqual(3);
+            this.tracer.RelatedErrorEvents.Count.ShouldEqual(0);
+            this.tracer.RelatedWarningEvents.Count.ShouldEqual(1);

```

Comment:
I thought about that.  Then I'd make the messages string const and that gets tricky for those that have variables.  Considering most unit tests don't do the tracer of level of detail and we verify all the outputs I think this hits the mark for test coverage.

Reply:
Let's leave that message as a TODO for #650.

Reply:
@derrickstolee @wilbaker just to throw it out there I'm not sure how much we really want to do this.  There is an overhead with verifying individual messages.  I've done in the past and have mixed feelings on the benefit versus maintenance.  There are certain cases where it's worth doing but I wouldn't say do it as a blanket rule.

Reply:
> There are certain cases where it's worth doing but I wouldn't say do it as a blanket rule.

I agree, I don't think we have to do it here.  I was just throwing it out as an option to consider in case you thought it was one of those cases where it'd be worth it.